libifupdown/lifecycle.c

```diff
@@ -182,6 +230,11 @@ build_environment(char **envp[], const struct lif_execute_opts *opts, const stru
 
 		lif_environment_push(envp, envkey, (const char *) entry->data);
 	}
+
+	if (addresses != NULL)
+		lif_environment_push(envp, "IF_ADDRESSES", addresses);
```

Comment:
after `addresses` and `gateways` is copied to environment, we should `free()` it to prevent a memory leak.